microceph/ceph/nfs_test.go

```diff
@@ -0,0 +1,352 @@
+	r.On("RunCommand", []interface{}{
+		"rados", "ls", "--pool", ".nfs.metadata", "--all", "--create"}...).Return("ok", nil).Once()
+	r.On("RunCommand", []interface{}{
+		"ceph", "osd", "pool", "application", "enable", ".nfs", "cephfs"}...).Return("ok", nil).Once()
+	r.On("RunCommand", []interface{}{
+		"rados", "create", "--pool", ".nfs", "-N", clusterID, obj}...).Return("ok", nil).Once()
+
+	// addNodeToSharedGraceMgmtDb call
+	r.On("RunCommand", []interface{}{
+		"ganesha-rados-grace", "--cephconf", cephconf, "--pool", ".nfs", "--ns", clusterID, "--userid", userID, "add", hostname}...).Return("ok", nil).Once()
+
+	// startNFS call
+	r.On("RunCommand", "snapctl", "start", "microceph.nfs-ganesha", "--enable").Return("ok", nil).Once()
+
+	// patch processExec
+	processExec = r
+
+	nfs := &NFSServicePlacement{
+		ClusterID:    clusterID,
+		BindAddress:  "0.0.0.0",
+		BindPort:     2049,
+		V4MinVersion: 2,
+	}
+
+	// function call
+	err = EnableNFS(s.TestStateInterface, nfs, []string{})
+
+	assert.NoError(s.T(), err)
+}
+
+func (s *NFSSuite) TestDisableNFS() {
+	hostname, err := os.Hostname()
+	assert.NoError(s.T(), err)
+
+	original := constants.GetPathConst
+	defer func() { constants.GetPathConst = original }()
+
+	constants.GetPathConst = func() constants.PathConst {
+		return constants.PathConst{
+			ConfPath: s.Tmp,
+		}
+	}
+
+	ganeshaConfDir := filepath.Join(s.Tmp, "ganesha")
+	keyringPath := filepath.Join(ganeshaConfDir, "keyring")
+	cephConf := filepath.Join(ganeshaConfDir, "ceph.conf")
+	ganeshaConf := filepath.Join(ganeshaConfDir, "ganesha.conf")
+
+	// Create the files, and expect them to be deleted.
+	err = os.MkdirAll(ganeshaConfDir, 0744)
+	assert.NoError(s.T(), err)
+
+	files := []string{keyringPath, cephConf, ganeshaConf}
+	for _, file := range files {
+		_, err := os.Create(file)
+		assert.NoError(s.T(), err)
+		defer os.Remove(file)
+	}
+
+	u := api.NewURL()
+	state := &mocks.MockState{
+		URL:         u,
+		ClusterName: "foohost",
+	}
+	s.TestStateInterface = mocks.NewStateInterface(s.T())
+	s.TestStateInterface.On("ClusterState").Return(state)
+
+	r := mocks.NewRunner(s.T())
+
+	// stopNFS call
+	r.On("RunCommand", "snapctl", "stop", "microceph.nfs-ganesha", "--disable").Return("ok", nil).Once()
+
+	// removeNodeFromSharedGraceMgmtDb call
+	clusterID := "foo"
+	userID := fmt.Sprintf("nfs.%s.%s", clusterID, hostname)
+	r.On("RunCommand", []interface{}{
+		"ganesha-rados-grace", "--cephconf", cephConf, "--pool", ".nfs", "--ns", clusterID, "--userid", userID, "remove", hostname}...).Return("ok", nil).Once()
+
+	// DeleteClientKey call
+	clientUser := fmt.Sprintf("client.%s", userID)
+	r.On("RunCommand", "ceph", "auth", "del", clientUser).Return("ok", nil).Once()
+
+	// patch processExec
+	processExec = r
+
+	// function call
+	err = DisableNFS(context.Background(), s.TestStateInterface, clusterID)
+
+	assert.EqualError(s.T(), err, "no server certificate")
```

Comment:
I believe `no server certificate` is an error message from the uninitialized database. 
Would it be possible to mock the DB so we can get a clean test run?